### PR TITLE
Createed `userResource` context and some refactor

### LINKED ISSUE #9: Create UserResource context which will handle all the user's resources like playlists, liked videos etc.
## To be created 
- UserResource context

### PR DESCRIPTION
**Pull request by kjais1720**
PR for Issue #9 ( closes #9 )
## Overview
- Created `userResource` context which handles all the api calls of Playlists, History, likes and watchlater
- Added `Add to liked videos` and `Add to Watchlater` functionality in the video cards

## Preview
Preview link - https://crickstreams-git-feat-create-userresource-context-kjais1720.vercel.app/ 
Preview image - 
image

### CODE REVIEW

File: src/components/videoCard.jsx [hunk 1]
```diff
@@ -10,33 +12,86 @@ export function VideoCard({ video }) {
...
+  } = resourcesApiStateEnums;
+
+  const {
+    userState: { isLoggedIn },
+    setShowAuthModal,
+  } = useAuth();
+
+  const isLiked = likedVideos.some((video) => video._id === id);
+  const isAddedToWatchlater = watchlater.some((video) => video._id === id);
+
```
Comment (raman1999) [inline src/components/videoCard.jsx:35]: just a suggestion , can make isIteminArray separate function for this

File: src/contexts/userResources/reducer.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  REMOVE_FROM_PLAYLIST: "REMOVE_FROM_PLAYLIST",
+  ADD_TO_HISTORY: "ADD_TO_HISTORY",
+  REMOVE_FROM_HISTORY: "REMOVE_FROM_HISTORY",
+  CLEAR_HISTORY: "CLEAR_HISTORY",
+  ADD_TO_LIKES: "ADD_TO_LIKES",
+  REMOVE_FROM_LIKES: "REMOVE_FROM_LIKES",
+  ADD_TO_WATCHLATER: "ADD_TO_WATCHLATER",
+  REMOVE_FROM_WATCHLATER: "REMOVE_FROM_WATCHLATER",
+};
+
```
Comment (raman1999) [inline src/contexts/userResources/reducer.js:17]: good to see you are using constants for this

**Review by raman1999 — APPROVED** [review]
well written code, good logical thinking done in making all functionalities, looks clean

File: src/components/videoCard.jsx (hunk 1 above)
Comment (kjais1720) [reply src/components/videoCard.jsx:35]: Thanks for the suggestion, will try to implement it

**Review by shubhamkumar648 — APPROVED** [review]
code looks clean good use of  function meaning convention

**Pull request merged**